ui/desktop/src/components/sessions/SessionListView.tsx

```diff
@@ -161,6 +162,50 @@ interface SearchContainerElement extends HTMLDivElement {
   _searchHighlighter: SearchHighlighter | null;
 }
 
+/**
+ * Determines if a session was created by the scheduler system.
+ * Uses dual detection logic to handle both current and legacy scheduler sessions.
+ * 
+ * Detection methods:
+ * 1. Primary: Check for non-null schedule_id metadata (current scheduler sessions)
+ * 2. Secondary: Pattern matching for timestamp-based session IDs (legacy fallback)
+ * 
+ * @param session - The session object to analyze
+ * @returns true if the session appears to be scheduler-generated
+ */
+// Optimized for performance with early returns and cached regex
+// Regex pattern for scheduler-generated timestamp IDs (YYYYMMDD_HHMMSS format)
+const SCHEDULER_TIMESTAMP_PATTERN = /^\d{8}_\d{6}$/;
+export const isSchedulerSession = (session: Session): boolean => {
```

Comment:
this function still reads vibe coded. why would we need to check anything but .schedule_id